its/src/test/java/com/sonar/it/scanner/msbuild/sonarcloud/CloudUtils.java

```diff
@@ -67,19 +69,21 @@ public static BuildResult runBeginStep(Path projectDir, String projectKey, Prope
     return result;
   }
 
+  @Deprecated // Use AnalysisContext instead
   public static BuildResult runEndStep(Path projectDir) {
     var result = ScannerCommand.createEndStep(ScannerClassifier.NET_FRAMEWORK, CloudConstants.SONARCLOUD_TOKEN, projectDir).execute(null);
     assertTrue(result.isSuccess());
     waitForTaskProcessing(result.getLogs());
     return result;
   }
 
+  // ToDo: SCAN4NET-10 will move/deprecate/remove this in favor of BuildCommand
   public static void runBuild(Path projectDir) {
     var result = TestUtils.runDotnetCommand(projectDir, "build", "--no-incremental");
     assertThat(result.isSuccess()).isTrue();
   }
 
-  private static void waitForTaskProcessing(String logs) {
+  public static void waitForTaskProcessing(String logs) {
```

Comment:
NIT:
Line `var request = HttpRequest.newBuilder(uri).header("Authorization", "Bearer " + System.getenv("SONARCLOUD_PROJECT_TOKEN")).build();` can be moved next to `var client` outside of the loop.

Reply:
I'll do this in another PR to save a CI re-run